# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/ipc/SocketProcessBridgeChild.h
# Commit: f18e61be455d
# Full Hash: f18e61be455d493f10a2cc651cd254859894e02e
# Author: Byron Campen [:bwc] <docfaraday@gmail.com>
# Date: 2019-02-26 03:52:35
# Regressor Bug: 1521879
# File Overlap Count: 1
# Description:
#   Bug 1521879 - Part 1: IPC-based MediaTransport implementation r=mjf
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D17273
# ==============================================================================

diff -r 8136a17fbec9 -r f18e61be455d netwerk/ipc/SocketProcessBridgeChild.h
--- a/netwerk/ipc/SocketProcessBridgeChild.h	Mon Feb 25 21:48:20 2019 +0000
+++ b/netwerk/ipc/SocketProcessBridgeChild.h	Mon Feb 25 21:50:42 2019 +0000
@@ -22,10 +22,10 @@
   NS_DECL_ISUPPORTS
   NS_DECL_NSIOBSERVER
 
-  static bool Create(Endpoint<PSocketProcessBridgeChild>&& aEndpoint);
   static already_AddRefed<SocketProcessBridgeChild> GetSingleton();
-  static void EnsureSocketProcessBridge(std::function<void()>&& aOnSuccess,
-                                        std::function<void()>&& aOnFailure);
+  typedef MozPromise<RefPtr<SocketProcessBridgeChild>, nsCString, false>
+      GetPromise;
+  static RefPtr<GetPromise> GetSocketProcessBridge();
 
   mozilla::ipc::IPCResult RecvTest();
   void ActorDestroy(ActorDestroyReason aWhy) override;
@@ -34,8 +34,12 @@
   bool Inited() const { return mInited; };
   ProcessId SocketProcessPid() const { return mSocketProcessPid; };
 
+  dom::PMediaTransportChild* AllocPMediaTransportChild() override;
+  bool DeallocPMediaTransportChild(dom::PMediaTransportChild* aActor) override;
+
  private:
   DISALLOW_COPY_AND_ASSIGN(SocketProcessBridgeChild);
+  static bool Create(Endpoint<PSocketProcessBridgeChild>&& aEndpoint);
   explicit SocketProcessBridgeChild(
       Endpoint<PSocketProcessBridgeChild>&& aEndpoint);
   virtual ~SocketProcessBridgeChild();